supportedBackends/google/pipelines/common/src/main/scala/cromwell/backend/google/pipelines/common/PipelinesApiAsyncBackendJobExecutionActor.scala

```diff
@@ -658,20 +659,20 @@ class PipelinesApiAsyncBackendJobExecutionActor(override val standardParams: Sta
   }
 
   override def mapCommandLineWomFile(womFile: WomFile): WomFile = {
-    womFile.mapFile(value =>
+    womFile.mapFile { value =>
       (getPath(value), asAdHocFile(womFile)) match {
         case (Success(gcsPath: GcsPath), Some(adHocFile)) =>
           // Ad hoc files will be placed directly at the root ("/cromwell_root/ad_hoc_file.txt") unlike other input files
           // for which the full path is being propagated ("/cromwell_root/path/to/input_file.txt")
           workingDisk.mountPoint.resolve(adHocFile.alternativeName.getOrElse(gcsPath.name)).pathAsString
-        case (Success(gcsPath: GcsPath), _) =>
-          workingDisk.mountPoint.resolve(gcsPath.pathWithoutScheme).pathAsString
+        case (Success(path @ ( _: GcsPath | _: HttpPath)), _) =>
```

Comment:
Is this pattern suggesting that `GcsPath` and `HttpPath` should inherit some shared `trait`?

Otherwise we're going to have to remember to append to all of these cases whenever a new FS like this is added (S3Path...?)

Reply:
(or `DosPath`...)

Reply:
Or FTP in the very near future (@Horneth )

Reply:
If N &gt; 2 I would agree. *Demo*DosPath below is currently doing something a little different and I'm not sure what's brewing for FTP.

Reply:
Hmm are we talking about something like `trait PathSupportedByPipelinesAPI` and have GcsPath etc extend it ?
I'm not totally convinced that would make things easier especially if filesystems impl and their respective `Path` are in separate isolated subprojects.